Library/Homebrew/hardware.rb

```diff
@@ -29,10 +29,11 @@ def optimization_flags
           native:             arch_flag("native"),
           ivybridge:          "-march=ivybridge",
           sandybridge:        "-march=sandybridge",
+          westmere:           "-march=westmere",
           nehalem:            "-march=nehalem",
           core2:              "-march=core2",
           core:               "-march=prescott",
-          arm_vortex_tempest: "",
+          arm_vortex_tempest: "", # TODO: -mcpu=apple-m1 when we've patched all our GCCs to support it
```

Comment:
Some earlier versions of Xcode may  barf on this too.

Reply:
In terms of LLVM: added in LLVM 13, though we don't support versioned LLVM as a compiler yet.

That would equate to Xcode 13.3 but it's very possible Apple had it in their internal Clang earlier. If we find any of Xcode 12.2-13.2 don't support it, we could probably swap it on shim level to `-mcpu=apple-a13` or something (not quite the same but close enough). We don't support 12.2-12.4 in the CLT so we could drop that on the Xcode side should `-mcpu=apple-m1` only be a problem there.

Reply:
Yea, Apple clang supported the flag earlier than upstream LLVM did IIRC. ("earlier" in a corresponding upstream LLVM version sense -- I think it was the latest released version of Xcode when upstream LLVM 13 dropped.)

NBD to enable this flag as-is as soon as we have sufficient GCC support -- we can fix any problems that come up as they emerge.

Reply:
I checked Apple's OSS LLVM fork but it wasn't there prior to it landing in LLVM so it seems they kept it in their private builds initially.

I can easily test with some CLT versions when the time arrives. The ones to test really are Xcode/CLT 12.2, 12.5 and 13.0 as those are the main big Clang branches, narrowing down further if necessary.